internal/services/monitor/monitor_diagnostic_setting_resource_test.go

```diff
@@ -131,7 +127,46 @@ func TestAccMonitorDiagnosticSetting_storageAccount(t *testing.T) {
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
 				check.That(data.ResourceName).Key("storage_account_id").Exists(),
-				check.That(data.ResourceName).Key("metric.#").HasValue("1"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccMonitorDiagnosticSetting_storageAccountTarget(t *testing.T) {
```

Comment:
Since this test isn't testing `storageAccountTarget` but rather updating `enabled_metric` could we change the name to: 

```suggestion
func TestAccMonitorDiagnosticSetting_updateEnabledMetric(t *testing.T) {
```

And since it looks like we can have multiple `enabled_metric`s, can we test that too while updating. Going from one to two and back to one again